financial-connections/src/main/java/com/stripe/android/financialconnections/features/generic/GenericScreen.kt

```diff
@@ -0,0 +1,345 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.onGloballyPositioned
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import com.stripe.android.financialconnections.features.common.IconSize
+import com.stripe.android.financialconnections.features.common.ShapedIcon
+import com.stripe.android.financialconnections.ui.FinancialConnectionsPreview
+import com.stripe.android.financialconnections.ui.LocalImageLoader
+import com.stripe.android.financialconnections.ui.TextResource
+import com.stripe.android.financialconnections.ui.components.AnnotatedText
+import com.stripe.android.financialconnections.ui.components.FinancialConnectionsButton
+import com.stripe.android.financialconnections.ui.sdui.fromHtml
+import com.stripe.android.financialconnections.ui.theme.FinancialConnectionsTheme.colors
+import com.stripe.android.financialconnections.ui.theme.FinancialConnectionsTheme.typography
+import com.stripe.android.financialconnections.ui.theme.Layout
+import com.stripe.android.uicore.image.StripeImage
+import androidx.compose.ui.Alignment as ComposeAlignment
+
+@Preview
+@Composable
+internal fun GenericScreenPreview(
+    @PreviewParameter(GenericScreenPreviewParameterProvider::class) state: GenericScreenState
+) {
+    FinancialConnectionsPreview {
+        Surface(color = colors.backgroundSurface) {
+            GenericScreen(
+                state = state,
+                onClickableTextClick = {},
+                onSecondaryButtonClick = {},
+                onPrimaryButtonClick = {},
+            )
+        }
+    }
+}
+
+@Composable
+internal fun GenericScreen(
+    state: GenericScreenState,
+    onPrimaryButtonClick: () -> Unit,
+    onSecondaryButtonClick: () -> Unit,
+    onClickableTextClick: (String) -> Unit,
+) {
+    val density = LocalDensity.current
+
+    var containerHeight by remember { mutableStateOf(0.dp) }
+    var footerHeight by remember { mutableStateOf(0.dp) }
+    var contentHeight by remember { mutableStateOf(0.dp) }
+
+    val spacing by remember {
+        derivedStateOf {
+            ((containerHeight - footerHeight - contentHeight) / 2).takeIf {
+                state.screen.options?.verticalAlignment == VerticalAlignment.Centered && it > 0.dp
+            }
+        }
+    }
+
+    Layout(
+        bodyPadding = PaddingValues(vertical = 16.dp),
+        inModal = state.inModal,
+        footer = state.screen.footer?.let {
+            {
+                GenericFooter(
+                    payload = it,
+                    modifier = Modifier.onGloballyPositioned {
+                        with(density) {
+                            footerHeight = it.size.height.toDp()
+                        }
+                    },
```

Comment:
Since this is used a few times: Can we create a Modifier extension method here, like `Modifier.onHeightDetermined(action: (Dp) -> Unit)`?

Reply:
removed centering logic for this PR, let's revisit it as soon as needed!